### PR TITLE
Tags: New component for managing multiple tags

### PR DESCRIPTION
**Pull request by iammuttaqi**
**Tags: New component for managing multiple tags**

Web capture_24-11-2023_22438_localhost

### CODE REVIEW

File: src/View/Components/Tags.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                <!-- LABEL -->
+                <label class="label label-text pt-0 font-semibold" for="{{ $uuid }}">{{ $label }}</label>
+
+                <!-- DIV AS INPUT -->
+                <div class="input input-primary relative flex h-fit w-full max-w-full place-items-center justify-between [outline:none!important]">
+                    <div class="py-1">
+                        <template :key="index" x-for="(tag, index) in tags">
+                            <div class="my-1 mr-2 inline-block cursor-pointer space-x-2 rounded bg-primary/20 px-2 py-1 transition-all hover:bg-primary/40">
+                                <span x-text="tag"></span>
+                                <svg class="inline-block h-4 w-4 stroke-current transition-all hover:text-red-500" fill="none" viewBox="0 0 24 24" x-on:click="remove(index)" xmlns="http://www.w3.org/2000/svg">
```
Comment (robsontenorio) [inline src/View/Components/Tags.php:75]: You can use here the internal x-mary-icon for “x” icon

File: src/View/Components/Tags.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                    <path d="M6 18L18 6M6 6l12 12" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></path>
+                                </svg>
+                            </div>
+                        </template>
+
+                        <!-- INPUT -->
+                        <input class="input input-bordered input-ghost input-sm my-1 rounded transition-all focus:input-primary" id="{{ $uuid }}" placeholder="{{ $attributes->whereStartsWith('placeholder')->first() }}" type="text" x-model="tag" x-on:click.outside="clear()" x-on:keydown.enter.prevent="push()" x-on:keyup.prevent="if (event.key === ',') { push() }" />
+                    </div>
+
+                    <svg class="inline-block h-6 w-6 min-w-fit cursor-pointer stroke-current transition-all hover:text-red-500" fill="none" viewBox="0 0 24 24" x-on:click="clearAll()" xmlns="http://www.w3.org/2000/svg">
```
Comment (robsontenorio) [inline src/View/Components/Tags.php:85]: Same here.

**Comment by iammuttaqi** [comment]
@robsontenorio PR updated

**Comment by robsontenorio** [comment]
@iammuttaqi 

When initial list contains a number it fails

```php
public array $tags = ['a', 11, 'c'];
```

image
image


When I click this area it clears the list. Are you able to put clear icon right side, just like Choices?

Captura de Tela 2023-11-27 às 13 01 52

**Comment by iammuttaqi** [comment]
@robsontenorio I couldn't find any issues, and it seems like everything is working as expected...

**Comment by iammuttaqi** [comment]
@robsontenorio updated something. please check.

**Comment by robsontenorio** [comment]
@iammuttaqi What was the bug?

**Comment by iammuttaqi** [comment]
@robsontenorio it was still taking spaces

image

**Comment by davidcabcab** [comment]
Any chance to use a tags list declared into a blade view instead using wire:model ?

This would be useful when iterating on a parent component, for example table rows. 

In this situation, tags would be accessible via  @php directly in the blade view. 

Unfortunately, < x-tags > seems to only read tags from the binded livewire variable.

Capture d’écran 2024-02-24 à 22 22 13

**Comment by robsontenorio** [comment]
The maryUI components are made for Livewire 3. Everything can be handled using Livewire 3.

For now I don't have plans to support components on plain blade views.

**Comment by robsontenorio** [comment]
Here a tricky to inject “external variables” into scopes. 

```
@scope(“xxx”, $var1. $var2, $var3)
```

**Comment by davidcabcab** [comment]
> And yes, `wire:model` is the way.
> 
> Here is another trick when you need to inject external vars into scopes
> 
> ```
> @scope('xxxx', $mainVar, $var2, $var3)
> ```

That's brilliant, it helps a lot thanks !

**Pull request merged**